components/ai_chat/browser/ai_chat_tab_helper.cc

```diff
@@ -721,10 +737,11 @@ void AIChatTabHelper::OnAPIStreamDataComplete(
   }
 
   if (!success) {
-    // TODO(petemill): show error state separate from assistant message
-    AddToConversationHistory(ConversationTurn{
-        CharacterType::ASSISTANT, ConversationTurnVisibility::VISIBLE,
-        l10n_util::GetStringUTF8(IDS_CHAT_UI_API_ERROR)});
+    SetAPIError(mojom::APIError::ConnectionIssue);
+
+    if (net::HTTP_TOO_MANY_REQUESTS == result.response_code()) {
+      SetAPIError(mojom::APIError::RateLimitReached);
+    }
```

Comment:
how about an `else` instead of calling SetAPIError twice?

Reply:
done